lib/galaxy/datatypes/binary.py

```diff
@@ -2168,6 +2168,29 @@ def sniff(self, filename):
             return f.read(8) == self._magic
 
 
+class RMA6(Binary):
+    """
+    Class describing an RMA6 (MEGAN6 read-match archive) file
+    >>> from galaxy.datatypes.sniff import get_test_fname
+    >>> fname = get_test_fname('diamond.rma')
```

Comment:
The new datatype tests requires the test data to end with `file_ext`, if you rename `'diamond.rma'` to `'diamond.rma6'` the test file will be uploaded, check that it will be sniffed correctly (in the presence of all other known datatypes) and downloaded again to verify the file is still intact.
(And I'm planning to extend this to all download all generated metadata)